### ISSUE TITLE
Order sync "stops working" / cron not running?

### ISSUE THREAD

**Issue by skwp** [reverbdotcom/reverb-magento#160]
This has happened to two implementations so far. I'm not sure about the root cause, but edting app/code/community/Reverb/ReverbSync/Helper/Orders/Retrieval/Creation.php to change the 1440 minute setting to a bigger number seems to help.

Hypothesis:
1. cron stops running at some point (maybe sync is temporarily turned off by customer)
2. cron resumes, but 1440 minutes is not big enough to fetch the missing orders

I believe to handle this properly we'll need to log the time of last sync and fetch from there plus some additional overlap (say 12 hours) to be totally sure we get everything.

It seems like if this happens new orders also don't come in, which is strange 1440mins is 24 hours so it's not clear why the customer would be missing today's orders

**Comment by skwp** [comment]
At: 11-16, 13:43:27.368 CST, the app polled for /orders/selling/awaiting_shipment?updated_start_date=2015-11-15T19:43:25-00:00

19:43 UTC  = 1:43 PM CST

According to this, it is correctly polling for things updated since 1 day ago

**Comment by skwp** [comment]
I think the actual problem may have to do with what endpoint we're polling. Right now /awaiting_shipment means literally things awaiting shipment which means it's not including local pickup orders. 

That does not seem like the very root cause of this bug but might be part of why people think orders are not coming through

**Comment by skwp** [comment]
I suspect some customers do not have a correctly functioning cron. By the time they realize it, they hit the sync button but the sync doesn't reach far back enough.

**Comment by skwp** [comment]
I have a feeling cron is running unreliably; I don't think it's working on my test instance anymore. cc @dunagan5887

**Comment by skwp** [comment]
Definitely not running :/ - when I manually invoke it I also get no output. However I can see this..maybe there's a deadlock of some kind

ps aux | grep cron | grep daemon
32039  0.0  0.0  64528  1532 pts/3    T    Aug24   0:00 su daemon -s /bin/sh -c /home/bitnami/apps/magento/htdocs/cron.sh cron.php -mdefault 1

**Comment by dunagan5887** [comment]
By test instance are you referring to bitnami@ec2-54-157-249-149.compute-1.amazonaws.com

**Comment by skwp** [comment]
yeah that's the one

**Comment by skwp** [comment]
I wiped out everything in var/log to get a clean slate but not seeing new logs just yet

**Comment by skwp** [comment]
I have new logs from clicking the sync button manually but the cron is definitely dead. This might be bitnami specific but I have other customers reporting the same issue

**Comment by skwp** [comment]
In my latest debugging of the cron on at least our instance, I'm trying to manually invoke it like this:
`/opt/bitnami/php/bin/php /opt/bitnami/apps/magento/htdocs/cron.php -mdefault 1`

I am expecting that my order syncs will be triggered but those log files don't show any activity

**Comment by dunagan5887** [comment]
I'm thinking this might be due to an issue with the lock files, I'll check out the server now

**Comment by dunagan5887** [comment]
Not sure how this happened, but the cron was no longer set to retrieve orders from Reverb, that may have happened when we staggered the creation/update cron processes. I have fixed that now

As for the dev server, I don't think the crontab entry being used was valid; I have made a new crontab entry and the Magento cron is running now. Additionally there were permissions issues preventing the bitnami user from writing to the log files in var/log; I have resolved the permissions issues, and the curl requests are now being logged.
